crates/parser/src/grammar/patterns.rs

```diff
@@ -65,14 +65,18 @@ fn pattern_single_r(p: &mut Parser, recovery_set: TokenSet) {
         //     match 92 {
         //         0 ... 100 => (),
         //         101 ..= 200 => (),
-        //         200 .. 301=> (),
+        //         200 .. 301 => (),
+        //         302 .. => (),
         //     }
         // }
         for &range_op in [T![...], T![..=], T![..]].iter() {
             if p.at(range_op) {
                 let m = lhs.precede(p);
                 p.bump(range_op);
-                atom_pat(p, recovery_set);
+                if !p.at(T![=>]) {
+                    // not a range pat like `302 .. => ()`
```

Comment:
Rather than leaving a comment here, better to split the tests:

```
if p.at(T![=>]) {
    // test half_open_range_pat
    // fn f() { let x .. = 1; }
} else {
    atom_pat(p, recovery_set);   
}
```

Reply:
You also found a bug with a test of that form! Fixed. Most likely would be seen in user code as `if let 1.. = (...)` which was not handled properly and is now.